### ISSUE TITLE
delete still possible when using the "read" permission from the examples?

### ISSUE THREAD

**Issue by joepstender** [curiosum-dev/permit#35]
When testing these permissions with a live_view from the examples, I still seem to be able to delete a blogpost as a regular user. All other permissions seem to be working fine, but deleting is still possible.

```elixir
defmodule TestingPermit.Permissions do
  use Permit.Ecto.Permissions, actions_module: Permit.Phoenix.Actions

  alias TestingPermit.Accounts.User
  alias TestingPermit.Content.Blog

  def can(%User{role: "admin"} = _user) do
    permit()
    |> all(Blog)
  end

  def can(%User{role: "user"} = _user) do
    permit()
    |> read(Blog)
  end

  def can(_user), do: permit()
end
```
```elixir
  defmodule TestingPermitWeb.BlogLive.Index do
  use TestingPermitWeb, :live_view

  alias TestingPermit.Content
  alias TestingPermit.Content.Blog

  use Permit.Phoenix.LiveView,
    authorization_module: TestingPermit.Authorization,
    resource_module: Blog

  @impl true
  def fetch_subject(_socket, session) do
    TestingPermit.Accounts.get_user_by_session_token(session["user_token"])
  end

  @impl true
  def handle_unauthorized(_action, socket) do
    {:halt,
     socket
     |> put_flash(:error, "You are not authorized to perform this action.")
     |> push_redirect(to: "/blogs")}
  end

  @impl true
  def mount(_params, _session, socket) do
    {:ok, stream(socket, :blogs, Content.list_blogs())}
  end

  @impl true
  def handle_params(params, _url, socket) do
    {:noreply, apply_action(socket, socket.assigns.live_action, params)}
  end

...

  @impl true
  def handle_event("delete", %{"id" => id}, socket) do
    blog = Content.get_blog!(id)
    {:ok, _} = Content.delete_blog(blog)

    {:noreply, stream_delete(socket, :blogs, blog)}
  end
end
```

**Comment by vincentvanbush** [comment]
@joepstender, thanks for writing and apologies for a late reply.
I am investigating into this, in the meantime though, would you be so kind to `inspect` the user record you're using this example with and paste it here?

**Comment by vincentvanbush** [comment]
@joepstender Returning with an answer on this: this is, indeed, technically correct behaviour for now, albeit admittedly poorly documented, and calling for improvement - thank you again for spotting this.

Since `:delete` is not performed via an URL change (and, hence, a change of the `:live_action` assign), authorization is not performed automatically by `Permit.Phoenix.LiveView`, and you would need to manually check for authorization to `:delete` via `can(user) |> delete?(record)`.

A possible improvement might be not only to authorize based on `:live_action` (hooking into the `handle_params` callback) but also hook into `handle_event` and `handle_info`.
This would require getting into https://github.com/curiosum-dev/permit_phoenix/blob/main/lib/permit_phoenix/live_view/authorize_hook.ex#L118 and adding event and info hooks triggering authorization.

I will explore this soon and let you know!

**Comment by vincentvanbush** [comment]
Referenced in curiosum-dev/permit_phoenix#11, closed via curiosum-dev/permit_phoenix#12. Thank you!
